src/main/resources/db/migration/V1__tables.sql

```diff
@@ -40,23 +40,37 @@ CREATE INDEX idx_object_log_old_object_id ON object_log (old_object_id) WHERE ol
 
 CREATE TABLE versions
 (
-    id                BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
-    session_id        TEXT   NOT NULL,
-    serial            BIGINT NOT NULL,
-    last_log_entry_id BIGINT NOT NULL,
-    snapshot_hash     TEXT CHECK (snapshot_hash ~ '^[0-9a-f]{64}$'),
-    delta_hash        TEXT CHECK (delta_hash ~ '^[0-9a-f]{64}$'),
-    snapshot_size     BIGINT,
-    delta_size        BIGINT,
-    created_at        TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
+    id                 BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    session_id         TEXT                     NOT NULL,
+    serial             BIGINT                   NOT NULL,
+    last_log_entry_id  BIGINT                   NOT NULL,
+    snapshot_file_name TEXT,
+    delta_file_name    TEXT,
+    snapshot_hash      TEXT CHECK (snapshot_hash ~ '^[0-9a-f]{64}$'),
+    delta_hash         TEXT CHECK (delta_hash ~ '^[0-9a-f]{64}$'),
+    snapshot_size      BIGINT,
+    delta_size         BIGINT,
```

Comment:
It might be nice to normalize these columns to have a separate `generated_files` table with `id, file_name, hash, size` and then have two FKs here for `snapshot_file_id` and `delta_file_id`. Requires some more joins but fewer `check` constraints.

Might also make it easier to handle generated files in a more consistent way, but not sure about that.